lib/puppet/provider/podman_secret/podman.rb

```diff
@@ -0,0 +1,114 @@
+require 'tempfile'
+require 'base64'
+require 'json'
+
+Puppet::Type.type(:podman_secret).provide(:podman) do
+  desc 'Podman secret provider'
+
+  commands podman: 'podman'
+
+  def exists?
+    podman('secret', 'inspect', resource[:name])
+    true
+  rescue Puppet::ExecutionFailure
+    false
+  end
+
+  def secret
+    return nil unless exists?
+
+    begin
+      output = execute_podman_command(['secret', 'inspect', '--showsecret', resource[:name]], capture_output: true)
```

Comment:
Is there any chance the output gets logged somewhere, even if we don't capture it?

Reply:
ACK :)